libbpf-rs/src/object.rs

```diff
@@ -225,6 +225,40 @@ impl OpenObject {
         self.map(name).unwrap().unwrap()
     }
 
+    /// Function that returns an Iterator over all available maps
+    /// in a given BPF [`Object`]
+    pub fn maps(&mut self) -> Result<&HashMap<String, OpenMap>> {
+        if !self.maps.is_empty() {
+            Ok(&self.maps)
```

Comment:
Not sure this logic is right. If user only accessed 1 map by name, then the iterator will only ever return 1 map even if there's more.